test/feeds.test.js

```diff
@@ -133,6 +134,88 @@ describe('test POST /feeds endpoint', () => {
+    const res = await request(app).put('/feeds/cache').send().set('Accept', 'application/json');
+    expect(res.status).toEqual(403);
+  });
+
+  it('should respond with a 403 status trying to clear cache when logged in as normal user', async () => {
+    login('Johannes Kepler', 'user1@example.com');
+    const res = await request(app).put('/feeds/cache').send().set('Accept', 'application/json');
+    expect(res.status).toEqual(403);
+  });
+
+  it('should respond with a 204 status trying to clear cache when logged in as admin user', async () => {
+    loginAdmin('Johannes Kepler', 'user1@example.com');
+    const res = await request(app).put('/feeds/cache').send().set('Accept', 'application/json');
+    expect(res.status).toEqual(204);
+  });
+
+  it('should respond with a 204 and all existing feed etag + lastModified be null', async () => {
+    const urlToId = (url) => hash(normalizeUrl(url));
+
+    const data = {
+      author: 'Post Author',
+      url: 'https://user.feed.com/feed.rss',
+      user: 'user',
+      id: urlToId('https://user.feed.com/feed.rss'),
+      link: 'https://user.feed.com/',
+      etag: 'etag',
+      lastModified: 'lastModified',
+    };
+
+    const data2 = {
+      author: 'Post Author2',
+      url: 'https://user2.feed.com/feed.rss',
+      user: 'user2',
+      id: urlToId('https://user2.feed.com/feed.rss'),
+      link: 'https://user2.feed.com/',
+      etag: 'etag',
+      lastModified: 'lastModified',
+    };
+
+    const data3 = {
+      author: 'Post Author3',
+      url: 'https://user3.feed.com/feed.rss',
+      user: 'user',
+      id: urlToId('https://user3.feed.com/feed.rss'),
+      link: 'https://user3.feed.com/',
+      etag: 'etag',
+      lastModified: 'lastModified',
+    };
+
+    const feedIds = await Promise.all([Feed.create(data), Feed.create(data2), Feed.create(data3)]);
+    const feeds = await Promise.all(feedIds.map(Feed.byId));
+
+    // Checking db against data
+    expect(feeds[0].etag).toBe('etag');
+    expect(feeds[1].etag).toBe('etag');
+    expect(feeds[2].etag).toBe('etag');
+    expect(feeds[0].lastModified).toBe('lastModified');
+    expect(feeds[1].lastModified).toBe('lastModified');
+    expect(feeds[2].lastModified).toBe('lastModified');
+
+    loginAdmin('Johannes Kepler', 'user1@example.com');
+    let res = await request(app).put('/feeds/cache').send().set('Accept', 'application/json');
+    expect(res.status).toBe(204);
+
+    // Checking db for updated values after put route
+    const clearedFeeds = await Promise.all(feedIds.map(Feed.byId));
+    expect(clearedFeeds[0].etag).toBe(null);
+    expect(clearedFeeds[1].etag).toBe(null);
+    expect(clearedFeeds[2].etag).toBe(null);
+    expect(clearedFeeds[0].lastModified).toBe(null);
+    expect(clearedFeeds[1].lastModified).toBe(null);
+    expect(clearedFeeds[2].lastModified).toBe(null);
+
+    // Make sure logging out as admin does not allow us to use the put route
+    logout();
```

Comment:
You've already tested this above, I don't think you need to do it again.

Reply:
removed